### PR TITLE
test: add benchmarks and benchmark helper functions

### PR DESCRIPTION
**Pull request by zachary-bailey**
**What type of PR is this?**

/kind test

**What this PR does / why we need it**:

This PR adds benchmarking helper functions to cse_helpers.sh and then utilizes them in pre / post / install-dependencies scripts.  These can be used to view benchmarking data while running ADO pipelines, and can be reviewed during the build or after. 

This data will later be used to programmatically evaluate the effect of a PRs code changes on the VHD Build Time.

**Which issue(s) this PR fixes**:

This code prevents the developer from not being able to easily evaluate how long certain parts of the VHD Build process take.

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_helpers.sh [hunk 1]
```diff
@@ -376,4 +376,44 @@ should_skip_nvidia_drivers() {
     should_skip=$(echo "$body" | jq -e '.compute.tagsList | map(select(.name | test("SkipGpuDriverInstall"; "i")))[0].value // "false" | test("true"; "i")')
     echo "$should_skip"
 }
+
+start_watch () {
+  capture_time=$(date +%s)
+  dt_start=$(date +%H:%M:%S)
```
Comment (ganeshkumarashok) [inline parts/linux/cloud-init/artifacts/cse_helpers.sh:382]: nit: rename to start_timestamp? "dt" is an unclear acronym

File: parts/linux/cloud-init/artifacts/cse_helpers.sh [hunk 2]
```diff
@@ -376,4 +376,44 @@ should_skip_nvidia_drivers() {
...
+}
+
+stop_watch () {
+
+  local current_time=$(date +%s)
+  local dt_end=$(date +%H:%M:%S)
```
Comment (ganeshkumarashok) [inline parts/linux/cloud-init/artifacts/cse_helpers.sh:388]: nit: rename to end_timestamp, for clarity too too?

Comment (zachary-bailey) [reply parts/linux/cloud-init/artifacts/cse_helpers.sh:388]: Will do for both!

File: parts/linux/cloud-init/artifacts/cse_helpers.sh [hunk 3]
```diff
@@ -376,4 +376,44 @@ should_skip_nvidia_drivers() {
...
+  echo "$benchmark"
+  echo
+  benchmarks+=("$benchmark")
+  benchmarks+=("$start")
+  benchmarks+=("$end")
+}
+
+show_benchmarks () {
+  echo
+  echo "Benchmarks:"
```
Comment (ganeshkumarashok) [inline parts/linux/cloud-init/artifacts/cse_helpers.sh:412]: Can we consolidate it to one line like "echo "\nBenchmark\n" by escaping the new lines? Will be shorter

**Pull request closed without merging**